tests/modules/images/regenerate-existing-images/background-process/perflab-background-job-test.php

```diff
@@ -0,0 +1,228 @@
+	}
+
+	/**
+	 * @covers ::create
+	 */
+	public function test_create_job() {
+		$job_data = array(
+			'post_id'          => 10,
+			'some_random_data' => 'some_random_string',
+		);
+
+		$job = perflab_create_background_job( 'test_job', $job_data );
+
+		$this->assertInstanceOf( Perflab_Background_Job::class, $job );
+	}
+
+	/**
+	 * @covers ::set_error
+	 * @covers ::get_attempts
+	 */
+	public function test_set_error() {
+		$error      = new WP_Error();
+		$error_data = array(
+			'test_error_data' => 'descriptive_infomation',
+		);
+		$error->add_data( $error_data, 'perflab_job_failure' );
+		$job_data = array(
+			'post_id'          => 10,
+			'some_random_data' => 'some_random_string',
+		);
+
+		$this->job = perflab_create_background_job( 'tecdcdcdst', $job_data );
+		$this->job->set_error( $error );
+
+		$error_metadata = get_term_meta( $this->job->get_id(), 'perflab_job_errors', true );
+		$attempts       = $this->job->get_attempts();
+
+		$this->assertSame( $error_data, $error_metadata );
+		$this->assertEquals( 1, $attempts );
+	}
+
+	/**
+	 * @covers ::should_run
+	 */
+	public function test_job_should_run_for_job() {
+		$this->job = perflab_create_background_job( 'test' );
+
+		$run = $this->job->should_run();
+		$this->assertTrue( $run );
+	}
+
+	/**
+	 * @covers ::should_run
+	 */
+	public function test_job_should_not_run_for_completed_job() {
+		$this->job = perflab_create_background_job( 'test' );
+
+		// Mark job as complete.
+		$this->job->set_status( 'perflab_job_complete' );
+		$run = $this->job->should_run();
+
+		$this->assertFalse( $run );
+	}
+
+	/**
+	 * @covers ::lock
+	 * @covers ::unlock
+	 */
+	public function test_lock_unlock() {
+		$this->job = perflab_create_background_job( 'test' );
+		$time      = time();
+		$this->job->lock( $time );
+		$lock_time = get_term_meta( $this->job->get_id(), 'perflab_job_lock', true );
+		$this->assertSame( absint( $lock_time ), $time );
+		$this->job->unlock();
+		$lock_time = get_term_meta( $this->job->get_id(), 'perflab_job_lock', true );
+		$this->assertEmpty( $lock_time );
+	}
+
+	/**
+	 * @covers ::get_start_time
+	 */
+	public function test_get_start_time() {
+		$this->job = perflab_create_background_job( 'test' );
+		$time      = time();
+		$this->job->lock( $time );
+		$lock_time = get_term_meta( $this->job->get_id(), 'perflab_job_lock', true );
+		$this->assertSame( absint( $lock_time ), $time );
+
+		$start_time = $this->job->get_start_time();
+		$this->assertEquals( $start_time, $lock_time );
+		$this->assertEquals( $start_time, $time );
+		$this->assertEquals( $time, $lock_time );
+	}
+
+	public function test_max_attempts_filter() {
+		$job = perflab_create_background_job( 'test' );
+		add_filter( 'perflab_job_max_attempts_allowed', array( $this, 'max_attempts_10' ) );
+		$job->should_run();
+		remove_filter( 'perflab_job_max_attempts_allowed', array( $this, 'max_attempts_10' ) );
+		$filter_ran = property_exists( $this, 'attempt_filter_ran' );
```

Comment:
Please don't set a property that doesn't exist (relevant below where this is set). We should declare the property as a private property of the test class and then check for its actual value here.

Or even better, it doesn't even need to be a class property: You can use closures to use a variable within this test method. That also avoids the need for having methods just to use as "once off throwaway" filters.

Why return 10 here? Isn't the default 3, so this should work even with that default value?

Last but not least: Filters added in a test don't need to removed, the WP test suite takes care of it.

```suggestion
		$job        = perflab_create_background_job( 'test' );
		$filter_ran = false;
		add_filter(
			'perflab_job_max_attempts_allowed',
			function( $value ) use ( &$filter_ran ) {
				$filter_ran = true;
				return $value;
			}
		);
		$job->should_run();
```